app/assets/javascripts/ui.js

```diff
@@ -198,38 +198,55 @@
     return true;
   });
 
-  // Adds a HTML
+  // Adds a new row
   $(document).on("focusout", "input[data-add-item-unless]", function () {
     var element = $(this);
     if (element.numericalValue() !== 0 && !$(element.data("add-item-unless")).hasClass("valid")) {
       if (element.data("toggled-by") === undefined || $(element.data("toggled-by")).is(':checked')) {
-        if (element.data("with")) {
-          var params = {};
-          $(element.data("with")).each(function () {
-            var paramName = $(this).data("parameter-name") || $(this).attr("id");
-            if (paramName !== null && paramName !== undefined) {
-              params[paramName] = $(this).val() || $(this).html();
+        $(document).one('cocoon:after-insert', function(e, newRow) {
+          var fillItemWith = element.data("fill-item-with");
+          var attributesToFill = fillItemWith.split(',');
+          attributesToFill = attributesToFill.map(function(attr) {
+            return attr.split('=>');
+          });
+          attributesToFill.forEach(function(toFromPair) {
+            var [toField, from] = toFromPair;
+            var [scope, fromField] = from.split('@');
+            var scope = $(element).closest(scope) || $(element).closest(element.data("use-closest")) || $(document);
+            var fromField = $(scope).find(fromField);
+            if ($(toField).parent().hasClass('selector') && fromField.val() !== null ) {
+              $(newRow).find(toField).data('ui-selector').value(fromField.val());
+            } else {
+              var fromValue = (fromField.val().length < 1) || isNaN(fromField.val()) ? fromField.val() : fromField.numericalValue();
+              $(newRow).find(toField).val(fromValue);
             }
           });
-          element.data("params", $.param(params));
-        }
-        $.rails.handleRemote(element);
+        });
+        $('a[data-association-insertion-node]').click();
       }
     }
   });
 
-  // Nullify inputs if it filled
+  // Nullify sibling input
   $(document).on("keyup", "input[data-exclusive-nullify]", function () {
     var element = $(this);
-    var scope = $("html");
-    if (element.data("use-closest")) {
-      scope = element.closest(element.data("use-closest"));
-    }
     if (element.numericalValue() !== 0) {
-      scope.find(element.data("exclusive-nullify")).val('');
+      $(element).closest('tr').find('input'+element.data('exclusive-nullify')).val('');
```

Comment:
This should be 're-generified', (see old implementation that uses `data-use-closest` rather than `closest(tr)`). This also begs the question of why it was rewritten.